<?xml version="1.0" encoding="UTF-8"?>
<pd:ProcessDefinition xmlns:pd="http://xmlns.tibco.com/bw/process/2003" xmlns:ns2="http://three.co.uk/xsd/resource/common/commondefinitions/msf/messagecontext/v1" xmlns:BW="java://com.tibco.pe.core.JavaCustomXPATHFunctions" xmlns:ns1="http://www.tibco.com/namespaces/tnt/plugins/jms" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:ns4="http://www.tibco.com/pe/GenerateErrorActivity/InputSchema" xmlns:ns="http://www.tibco.com/pe/EngineTypes" xmlns:ns3="http://three.co.uk/xsd/resource/common/commondefinitions/msf/exception/v1" xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" xmlns:tib="http://www.tibco.com/bw/xslt/custom-functions" xmlns:pfx2="http://three.co.uk/wsdl/interface/product/productspecification/productspecificationmanagement/mcfcmanagement/v1" xmlns:ns5="http://three.co.uk/xsd/resource/common/commondefinitions/msf/exceptioninfo/v1" xmlns:ns6="http://www.tibco.com/namespaces/tnt/plugins/file" xmlns:ns7="http://three.co.uk/xsd/resource/common/commondefinitions/msf/inputmessagecontent/v1" xmlns:pfx="http://three.co.uk/wsdl/interface/managedentityabe/managemententityabe/submanagemententityabe/servicename/vn">
    <wsdl:import namespace="http://three.co.uk/wsdl/interface/product/productspecification/productspecificationmanagement/mcfcmanagement/v1" location="/SharedResources/WebServiceDefinitions/Interface/Product/ProductSpecification/ProductSpecificationManagement/MCFCManagement_v1.wsdl"/>
    <pd:name>IntegrationServices/Product/ProductSpecification/ProductSpecificationManagement/MCFCManagement_v1/LoadProductModel/Starter/JMSTrigger.process</pd:name>
    <pd:startName>JMS Queue Receiver</pd:startName>
    <pd:startX>0</pd:startX>
    <pd:startY>0</pd:startY>
    <pd:returnBindings>
        <pfx:OperationReply>
            <part1>
                <xsl:copy-of select="$Call-Main/root/ns1:OperationReply"/>
            </part1>
        </pfx:OperationReply>
    </pd:returnBindings>
    <pd:starter name="JMS Queue Receiver">
        <pd:type>com.tibco.plugin.jms.JMSQueueEventSource</pd:type>
        <pd:resourceType>ae.activities.JMSQueueEventSource</pd:resourceType>
        <pd:x>65</pd:x>
        <pd:y>83</pd:y>
        <config>
            <PermittedMessageType>Text</PermittedMessageType>
            <SessionAttributes>
                <transacted>false</transacted>
                <acknowledgeMode>1</acknowledgeMode>
                <maxSessions>1</maxSessions>
                <destination>%%Destinations/Queues/LoadProductModel/LoadProductModelTrigger%%</destination>
            </SessionAttributes>
            <ConfigurableHeaders>
                <JMSDeliveryMode>PERSISTENT</JMSDeliveryMode>
                <JMSExpiration>0</JMSExpiration>
                <JMSPriority>4</JMSPriority>
            </ConfigurableHeaders>
            <ConnectionReference>/SharedResources/Connections/JMS/MSF-JMS-INT-Connection.sharedjmscon</ConnectionReference>
        </config>
        <pd:inputBindings/>
    </pd:starter>
    <pd:endName>End</pd:endName>
    <pd:endX>1258</pd:endX>
    <pd:endY>83</pd:endY>
    <pd:errorSchemas>
        <Schema0 wsMsgRef="pfx2:ExceptionInfo"/>
    </pd:errorSchemas>
    <pd:processVariables/>
    <pd:targetNamespace>http://xmlns.example.com/1355136884829</pd:targetNamespace>
    <pd:activity name="Audit End">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>841</pd:x>
        <pd:y>83</pd:y>
        <config>
            <processName>/UtilityProcesses/Audit/LogAuditEnd.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <response>
                    <NoResponse/>
                </response>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Audit Start">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>411</pd:x>
        <pd:y>83</pd:y>
        <config>
            <processName>/UtilityProcesses/Audit/LogAuditStart.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <payload>
                    <xsl:copy-of select="$JMS-Queue-Receiver/ns1:ActivityOutput/Body"/>
                </payload>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Call Main">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>526</pd:x>
        <pd:y>83</pd:y>
        <config>
            <processName>/IntegrationServices/Product/ProductSpecification/ProductSpecificationManagement/MCFCManagement_v1/LoadProductModel/Main/LoadProductMainProcess.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <CentralCatalogueServicesRequest>
                    <xsl:value-of select="$File-Poller/ns6:EventSourceOuputTextClass/fileContent/textContent"/>
                </CentralCatalogueServicesRequest>
                <SourceFileName>
                    <xsl:value-of select="$File-Poller/ns6:EventSourceOuputTextClass/fileInfo/fileName"/>
                </SourceFileName>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Catch All">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>110</pd:x>
        <pd:y>305</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <catchAll>true</catchAll>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Catch Exception">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>250</pd:x>
        <pd:y>445</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <fault>localname=Exception namespace=http://three.co.uk/xsd/resource/common/commondefinitions/msf/exception/v1</fault>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Create Exception Catch All">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>250</pd:x>
        <pd:y>313</pd:y>
        <config>
            <processName>/UtilityProcesses/Exceptions/CreateException_CatchAll.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <xsl:copy-of select="$_error/ns:ErrorReport"/>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Create Exception Receive Error">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>313</pd:x>
        <pd:y>189</pd:y>
        <config>
            <processName>/UtilityProcesses/Exceptions/CreateException_ReceiveError.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <xsl:copy-of select="$_error/ns:ErrorReport"/>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Create Message Context REPLY">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>687</pd:x>
        <pd:y>92</pd:y>
        <config>
            <processName>/UtilityProcesses/MessageContext/CreateMessageContext_REPLY.process</processName>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="InitialiseBoundary">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>185</pd:x>
        <pd:y>83</pd:y>
        <config>
            <processName>/UtilityProcesses/StartUp/InitialiseBoundary.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <InitialProcessName>
                    <xsl:value-of select="BW:getCurrentProcessName($_processContext/ns:ProcessContext/ProcessId)"/>
                </InitialProcessName>
                <ns7:InputMessageContent>
                    <xsl:copy-of select="$JMS-Queue-Receiver/ns1:ActivityOutput/Body"/>
                </ns7:InputMessageContent>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="InitialiseBoundary Error">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>68</pd:x>
        <pd:y>190</pd:y>
        <config>
            <processName>/UtilityProcesses/StartUp/InitialiseBoundary.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <InitialProcessName>
                    <xsl:value-of select="BW:getCurrentProcessName($_processContext/ns:ProcessContext/ProcessId)"/>
                </InitialProcessName>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Log Exception Catch All">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>390</pd:x>
        <pd:y>313</pd:y>
        <config>
            <processName>/UtilityProcesses/Exceptions/LogException.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <xsl:copy-of select="$Create-Exception-Catch-All/root/ns3:Exception"/>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Log Exception Catch Exception">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>389</pd:x>
        <pd:y>451</pd:y>
        <config>
            <processName>/UtilityProcesses/Exceptions/LogException.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <xsl:copy-of select="$Catch-Exception/ns3:Exception"/>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Log Exception Receive Error">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>490</pd:x>
        <pd:y>190</pd:y>
        <config>
            <processName>/UtilityProcesses/Exceptions/LogException.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <xsl:copy-of select="$Create-Exception-Receive-Error/root/ns3:Exception"/>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Set Ouput Context">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>983</pd:x>
        <pd:y>83</pd:y>
        <config>
            <processName>/UtilityProcesses/ContextResources/SetOutputContext.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <xsl:copy-of select="$Create-Message-Context-REPLY/root/ns2:MessageContext"/>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Trace End">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>1152</pd:x>
        <pd:y>83</pd:y>
        <config>
            <processName>/UtilityProcesses/Tracing/TraceEnd.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <processName>
                    <xsl:value-of select="BW:getCurrentProcessName($_processContext/ns:ProcessContext/ProcessId)"/>
                </processName>
                <body>
                    <xsl:value-of select="&quot;Starting JMS Trigger for Load Product Model. Trace log End&quot;"/>
                </body>
                <response>
                    <NoResponse/>
                </response>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Trace Start">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>295</pd:x>
        <pd:y>83</pd:y>
        <config>
            <processName>/UtilityProcesses/Tracing/TraceStart.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <processName>
                    <xsl:value-of select="BW:getCurrentProcessName($_processContext/ns:ProcessContext/ProcessId)"/>
                </processName>
                <body>
                    <xsl:value-of select="&quot;Starting JMS Trigger for Load Product Model Service Trace log Started&quot;"/>
                </body>
                <payload>
                    <xsl:copy-of select="$JMS-Queue-Receiver/ns1:ActivityOutput"/>
                </payload>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:transition>
        <pd:from>Audit Start</pd:from>
        <pd:to>Call Main</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Set Ouput Context</pd:from>
        <pd:to>Trace End</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Trace End</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Create Exception Receive Error</pd:from>
        <pd:to>Log Exception Receive Error</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Catch All</pd:from>
        <pd:to>Create Exception Catch All</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Create Exception Catch All</pd:from>
        <pd:to>Log Exception Catch All</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Catch Exception</pd:from>
        <pd:to>Log Exception Catch Exception</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Trace Start</pd:from>
        <pd:to>Audit Start</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Call Main</pd:from>
        <pd:to>Create Message Context REPLY</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Log Exception Receive Error</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>One Bend</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Create Message Context REPLY</pd:from>
        <pd:to>Audit End</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Audit End</pd:from>
        <pd:to>Set Ouput Context</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Log Exception Catch All</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>One Bend</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Log Exception Catch Exception</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>One Bend</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>InitialiseBoundary</pd:from>
        <pd:to>Trace Start</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>InitialiseBoundary Error</pd:from>
        <pd:to>Create Exception Receive Error</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>JMS Queue Receiver</pd:from>
        <pd:to>InitialiseBoundary</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>JMS Queue Receiver</pd:from>
        <pd:to>InitialiseBoundary Error</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-65536</pd:lineColor>
        <pd:conditionType>error</pd:conditionType>
    </pd:transition>
</pd:ProcessDefinition>